### ISSUE TITLE
Coverage date shows 0%files, 100% lines covered everywhere.

### ISSUE THREAD

**Issue by centy** [flutter/flutter-intellij#6404]
It looks like for some reasons data about coverage are not readed properly. I've checked wiht VSCode and it works there, so I guess coverage file is ok.

## Steps to Reproduce

Run unit tests with coverage. 
Get result like this:

![image](https://user-images.githubusercontent.com/581714/202820506-6814d847-24a2-4fe1-8d52-125c831ed073.png)

## Version info
[√] Flutter (Channel stable, 3.3.8, on Microsoft Windows [Version 10.0.19044.2251], locale pl-PL)
    • Flutter version 3.3.8 on channel stable at C:\Tools\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 52b3dc25f6 (10 days ago), 2022-11-09 12:09:26 +0800
    • Engine revision 857bd6b74c
    • Dart version 2.18.4
    • DevTools version 2.15.0

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0) 
    • Android SDK at C:\Users\jacek\AppData\Local\Android\sdk\
    • Platform android-33, build-tools 33.0.0
    • ANDROID_HOME = C:\Users\jacek\AppData\Local\Android\sdk\
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java       
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.1.0)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.1.32210.238
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[√] IntelliJ IDEA Community Edition (version 2022.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2022.2.3
    • Flutter plugin version 71.0.5
    • Dart plugin version 222.4345.14

[√] Connected device (1 available)
    • D6503 (mobile)                        • CB5A20P4UB    • android-arm    • Android 6.0.1 (API 23)

[√] HTTP Host Availability
    • All required HTTP hosts are available

Flutter framework as well as of the IntelliJ plugin.

**Comment by stevemessick** [comment]
Are you using Android Studio or IntelliJ? Can you share your lcov file?

**Comment by centy** [comment]
I have checked on both, same result.

**Comment by stevemessick** [comment]
Thanks for the response. The lcov file looks ok. Is your lcov file in a directory named `coverage`? If so, is it located at the root directory of your project? If you are using a monorepo project then it probably is not.

**Comment by stevemessick** [comment]
Interesting. That's actually the opposite of what I expected, but at least I have an idea of where to look for the problem. Thanks!

**Comment by SylphiaWindy** [comment]
Same for dart packages in one project.
Only tests in flutter module (ui app) works with coverage.
No 'test with coverage' option for tests in pure dart packages, and after coverage ran against flutter tests, all files in dart package showing '0% files, 100% lines covered' same as in the image centy posted.

**Comment by stevemessick** [comment]
@centy @SylphiaWindy I may need more information to make progress.

I made a sample application and added a plugin project to it. After running tests with coverage in both projects (from the same window) I got the coverage view I expected.

<img width="1394" alt="Screenshot 2023-01-06 at 1 23 54 PM" src="https://user-images.githubusercontent.com/8518285/211103071-72456360-4a80-4b8d-a286-3b532a1c7830.png">

The `mono_cov` project has a `test` directory but it is not shown in the screen shot. The right editor is `package/example/lib/main.dart` (highlighted) and the left editor is `lib/main.dart`.

Is this similar to your project structure? I first ran coverage at the top level `test` dir, then at the embedded `test` dir. The second time, IntelliJ asked me what to do with the results. I chose to combine them with the first run and got this result. Previously, I had chosen to replace the results, and got pretty much the same result that you reported.

When you run tests, do you get an option to combine or replace the results? If not, can you give me more detail of your project structure?

**Comment by centy** [comment]
I do not see coverage information for package on tree

pt., 6 sty 2023 o 22:35 stevemessick ***@***.***> napisał(a):

**Comment by stevemessick** [comment]
Correct. That's a separate problem. What you should notice is the red and green bars in the editors, which indicate that coverage data is available and has been processed.

Do you have an option to combine coverage results?

**Comment by SylphiaWindy** [comment]
@stevemessick 
Not exactly.
Why your packages folder have 'build', '.dart_tool' inside?

![structure](https://user-images.githubusercontent.com/11384750/211129518-1bed8568-4c43-42d9-a5bd-7e863dabf24f.png)

**Comment by stevemessick** [comment]
@SylphiaWindy I don't have a `packages` folder. It is `package` and actually should be called `plugin` because it is a plugin project. And that's why it has `.dart_tool` and `build`; I created it using the IDE, which opened a new window then did `pub get` (which created `.dart_tool`).

I believe the reason you and @centy are not seeing the same results as I do is because monorepo projects are tricky to construct. IntelliJ (or Android Studio) has to have the proper project structure. In mine, I did the extra steps required to make it work. It was simple: I used the Project Structure dialog (`File > Project Structure` menu) to import all the extra module definitions. Each Flutter project has a `*.iml` file that defines the IntelliJ module. Import all of them. If you don't do that then running tests in a sub-project won't add the results to the top-level project, and they won't be displayed correctly.

For more info about monorepo projects, see the [M69 release notes](https://groups.google.com/g/flutter-announce/c/on6eKqmhD3A), which has details about how to open a monorepo project so that all the IntelliJ modules are defined. Currently, I don't think there will be any new work done in this area, but I'm going to leave this open so others might more easily find it.

**Comment by SylphiaWindy** [comment]
I tried rearranging my project as M69 release notes with no luck, but I then realized that all my projects inside packages directory were pure dart projects.

```
/project_root
  .idea
  flutter_project
    .dart_tool
    lib
    test
  dart_project_1
    .dart_tool
    lib
    test
  README.md
```

There's no 'run with coverage'  at all for dart projects.

@stevemessick

**Comment by RomanPreis** [comment]
Hi! I've run into this bug as well, in my Flutter project, and received a reply after filing the bug in the Android Studio tracker to file it here.
I'm using AS: Giraffe | 2022.3.1 Patch 2, I've run tests with --coverage, my lcov is in coverage directory as it should be. I notice that in some cases it shows the percentage correctly (directory 'di', 0% files and 0% covered; 'network', 14% and 16% respectively, where I have written some tests) and in others like main.dart, internal directory and  gen directory(generated files by protoc) it show what seems like arbitrary numbers.
I do get an option to combine or replace test coverage info after running, but it didn't make any difference from what I can tell. I'd really appreciate any insight on this in case I'm missing something in the IDE settings
<img width="1552" alt="Screenshot 2023-10-05 at 21 00 06" src="https://github.com/flutter/flutter-intellij/assets/67924771/659617de-2a5b-499f-bb26-660ed97b45fb">
